test/tests/functional/pbs_qsub_opts_args.py

```diff
@@ -56,6 +56,17 @@ def setUp(self):
         self.fn = self.du.create_temp_file(body=script)
         self.qsub_cmd = os.path.join(
             self.server.pbs_conf['PBS_EXEC'], 'bin', 'qsub')
+        self.jobdir_root = "/tmp"
+        self.jobdir = []
+
+    def tearDown(self):
+        for mom in self.moms.values():
```

Comment:
Only your new test case needs this.  But putting it here means all 9 other tests will have to run through the setup and teardown.  Is it better to put this here or within the new test?

Reply:
I have to put it there as the test case could actually throw an exception on a self.assert() marking the test case failure, and this would be the only place where things can be cleaned up.

Reply:
Thanks for reminding me about the exception case.  Makes sense to me.

Reply:
What I usually do for things like this is to create a class variable which is False by default.  You set it to True at the start of the test.  If the variable is True, then you do something special in tearDown() and set it back to False.

Reply:
Good idea @bhroam : Let me do that.